cheevos/cheevos.c

```diff
@@ -1207,11 +1207,22 @@ void rcheevos_populate_menu(void* data)
 
    if (i == 0)
    {
-      menu_entries_append_enum(info->list,
+      if (!settings->arrays.cheevos_token[0])
```

Comment:
Is this checking if the `cheevos_token` is present in the `retroarch.cfg`?

IIRC after #6618 the `cheevos_token` will be stored after successfully loading achievements one time.

I'm wondering if the change here would cause a `Not logged in` message when you successfully contacted the server and the game really doesn't have achievements.

Reply:
It's leveraging this logic:
https://github.com/libretro/RetroArch/blob/762275b7940f62f41becf6ac6d56b00ff7ee8982/cheevos/cheevos.c#L2266-L2268

Which clears out the token on an invalid login. a token will also not exist if the player has never successfully logged in previously.

It's checking the variable in memory, but that is the field that gets read from/written to retroarch.cfg.

The user must be logged in to fetch achievements, so the token must exist before we can tell if a game really has achievements. If no achievements are found, we pause hardcore, but do not clear out the token. The token will be used for all future API calls (including logins) until it is no longer valid. #6618 _sets_ the token on successful login (and blanks out the _password_), so it will be set when the menu is opened.

The one case I didn't handle is if the login fails because the server cannot be contacted. In that case, we don't blank out the token, so it will still say "no achievements to display".

Reply:
> Which clears out the token on an invalid login.
this solves my doubt :+1: